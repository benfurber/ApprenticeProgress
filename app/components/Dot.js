// @flow

import React, { Component } from "react";
import { Animated, Easing } from "react-native";

import { colours } from "styles";

type Props = {
  delay: number,
};

class Dot extends Component<Props> {
  animatedValue: any;

  constructor() {
    super();
    this.animatedValue = new Animated.Value(0);
  }

  componentDidMount() {
    this.animate();
  }

  animate() {
    const { delay } = this.props;

    this.animatedValue.setValue(0);
    Animated.timing(this.animatedValue, {
      toValue: 1,
      delay,
      duration: 1800,
      easing: Easing.bounce,
    }).start(() => this.animate());
  }

  render() {
    const movingMargin = this.animatedValue.interpolate({
      inputRange: [0, 0.5, 1],
      outputRange: [0, 50, 0],
    });

    return (
      <Animated.View
        style={{
          backgroundColor: colours.red,
          borderRadius: 15,
          height: 30,
          width: 30,
          margin: 10,
          marginBottom: movingMargin,
        }}
      />
    );
  }
}

export { Dot };
