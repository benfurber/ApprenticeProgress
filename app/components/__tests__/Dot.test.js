// @flow

import React from "react";
import { shallow } from "enzyme";

import { Dot } from "../Dot";

describe("Dot", () => {
  it("has a valid snapshot", () => {
    const component = shallow(<Dot delay={0} />);

    expect(component).toMatchSnapshot();
  });
});
