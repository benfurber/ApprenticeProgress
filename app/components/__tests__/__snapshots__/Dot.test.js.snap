// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot has a valid snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dot
    delay={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "style": Object {
        "backgroundColor": "#ED3E44",
        "borderRadius": 15,
        "height": 30,
        "margin": 10,
        "marginBottom": 0,
        "width": 30,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "style": Object {
          "backgroundColor": "#ED3E44",
          "borderRadius": 15,
          "height": 30,
          "margin": 10,
          "marginBottom": 0,
          "width": 30,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
